src/Microsoft.VisualStudio.Composition/Reflection/TypeRef.cs

```diff
@@ -51,6 +51,11 @@ public class TypeRef : IEquatable<TypeRef>, IEquatable<Type>
         /// </summary>
         private ImmutableArray<TypeRef> baseTypes;
 
+        private static readonly Lazy<string> UnresolvableMetadataTokenString = new(() => Strings.UnresolvableMetadataToken);
```

Comment:
This adds overhead to acquiring the string, and you inevitably acquire the string anyway. It *does* store the retrieved string for reuse later, but it does so without concern for the culture set on the given thread, so you may return the wrong language.
There's another way to optimize the perf here so that you only retrieve the string in failure paths. I'll push the change.